tests/percy/iframepagenavigator.js

```diff
@@ -9,24 +9,50 @@ class IframePageNavigator extends PageNavigator {
    * @param {import('puppeteer').Page} page A Pupeteer Page
    * @param {string} siteUrl A url to the index of the site
    * @param {string} iframePage The name of the iframe page
+   * @param {string} defaultLocale default locale config of the site
    */
-  constructor(page, siteUrl, iframePage) {
+  constructor(page, siteUrl, iframePage, defaultLocale='en') {
     super();
     this._page = page;
     this._siteUrl = siteUrl;
     this._iframePage = iframePage;
+    this._defaultLocale = defaultLocale;
+
+    /**
+     * Locale param for the site's url on universal page
+     * 
+     * @type {string}
+     */
+    this._localeUniversalUrlPath = '';
+
+    /**
+     * Locale param for the site's url on vertical page
+     * 
+     * @type {string}
+     */
+    this._localeVerticalUrlPath = '&verticalUrl=';
+  }
+  
+  /**
+   * Sets locale param for the site's url based on given locale
+   * 
+   * @param {string} locale 
+   */
+  setCurrentLocale(locale) {
+    this._localeUniversalUrlPath = locale === this._defaultLocale? '' : '&verticalUrl=' + locale;
+    this._localeVerticalUrlPath = locale === this._defaultLocale? '&verticalUrl=' : '&verticalUrl=' + locale + "/";
   }
 
   async gotoUniversalPage(queryParams = {}) {
     const queryParamsString = getQueryParamsString(queryParams);
-    const url = `${this._siteUrl}/${this._iframePage}.html?${queryParamsString}`;
+    const url = `${this._siteUrl}/${this._iframePage}.html?${queryParamsString}${this._localeUniversalUrlPath}`;
     await this._page.goto(url);
     await waitTillHTMLRendered(this._page);
   }
 
   async gotoVerticalPage(vertical, queryParams = {}) {
     const queryParamsString = getQueryParamsString(queryParams);
-    const url = `${this._siteUrl}/${this._iframePage}.html?verticalUrl=${vertical}.html&${queryParamsString}`;
+    const url = `${this._siteUrl}/${this._iframePage}.html?${this._localeVerticalUrlPath}${vertical}.html&${queryParamsString}`;
```

Comment:
Rather than having a `_localeVerticalUrlPath`, would it be simpler to instead have a `_localePrefix` which would be '' for the default locale or something like 'es/' for Spanish?

Reply:
Alternatively we could check if a locale is present, and if so we have one way of defining the url, and if the url is not present, we have another way of defining the URL